neuromation/api/jobs.py

```diff
@@ -162,14 +171,43 @@ def __init__(self, core: _Core, config: _Config) -> None:
             # an error is raised for status >= 400
             return None  # 201 status code
 
-    async def save(self, id: str, image: RemoteImage) -> None:
+    async def save(
+        self,
+        id: str,
+        image: RemoteImage,
+        *,
+        progress: Optional[AbstractDockerImageProgress] = None,
+    ) -> RemoteImage:
         if not _is_in_neuro_registry(image):
             raise ValueError(f"Image `{image}` must be in the neuromation registry")
+
+        if progress is None:
+            progress = _DummyProgress()
+        progress.commit(JobProgressCommit(id, image))
+        is_committing = True
+        is_pushing = False
+
         payload = {"container": {"image": _as_repo_str(image)}}
         url = self._config.cluster_config.monitoring_url / f"{id}/save"
-        async with self._core.request("POST", url, json=payload):
-            # an error is raised for status >= 400
-            return None  # 201 status code
+        async with self._core.request("POST", url, json=payload) as resp:
+            async for chunk in resp.content:
+                obj = json.loads(chunk, encoding="utf-8")
+                if is_committing:
+                    step = _try_parse_job_commit_progress_step(obj)
+                    if step:
+                        progress.step(step)
+                        is_committing = not _is_commit_finshed(step)
+                else:
+                    if not is_pushing:
+                        # NOTE: here we don't expose our registry name in `src`
+                        src = LocalImage(f"{image.owner}/{image.name}", image.tag)
+                        progress.push(ImageProgressPush(src, dst=image))
+                        is_pushing = True
+                    step = _try_parse_image_progress_step(obj, image.tag)
+                    if step:
+                        progress.step(step)
+
+        return image
```

Comment:
You don't modify `image` in the method, what is the reason for returning argument as is?
Return nothing is enough.

Reply:
ok